integration/tests/bft/admin/blocklist/admin_command_blocklist_test.go

```diff
@@ -0,0 +1,46 @@
+package blocklist
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/onflow/flow-go/utils/unittest"
+)
+
+type AdminCommandBlockListTestSuite struct {
+	Suite
+}
+
+func TestAdminCommandBlockList(t *testing.T) {
+	suite.Run(t, new(AdminCommandBlockListTestSuite))
+}
+
+// TestAdminCommandBlockList ensures that the blocklist admin command works as expected. When a node is blocked via the admin blocklist command
+// the libp2p connection to that node should be pruned immediately and the connection gater should start to block incoming connection requests. This test
+// sets up 2 corrupt nodes a sender and receiver, the sender will send messages before and after being blocked by the receiver node via
+// the blocklist admin command. The receiver node is expected to receive messages like normal before blocking the sender, after blocking the sender
+// it should not receive any messages. The reason this test is conducted via two corrupt nodes is to empower the test logic to command one (corrupt) node to send a message and to examine the other (corrupt) node to check whether it has received the message.
+func (a *AdminCommandBlockListTestSuite) TestAdminCommandBlockList() {
+	// send some authorized messages indicating the network is working as expected
+	a.Orchestrator.sendAuthorizedMsgs(a.T())
+	unittest.RequireReturnsBefore(a.T(), a.Orchestrator.authorizedEventsReceivedWg.Wait, 5*time.Second, "could not receive authorized messages on time")
+	// messages with correct message signatures are expected to always pass libp2p signature verification and be delivered to the victim EN.
+	require.Equal(a.T(), int64(numOfAuthorizedEvents), a.Orchestrator.authorizedEventsReceived.Load(), fmt.Sprintf("expected to receive %d authorized events got: %d", numOfAuthorizedEvents, a.Orchestrator.expectedBlockedEventsReceived.Load()))
+
+	// after blocking node a.senderVN we should not receive any messages from that node. We wait for
+	// 500 milliseconds to reduce the small chance of a race condition between the time a node is blocked
+	// and the time the blocked node sends the first unauthorized message
+	a.blockNode(a.senderVN)
+	time.Sleep(500 * time.Millisecond)
+
+	// send unauthorized messages and sleep for 3 seconds to allow all requests to be processed
+	a.Orchestrator.sendExpectedBlockedMsgs(a.T())
+	time.Sleep(3 * time.Second)
```

Comment:
can we get rid of the sleep? Maybe have a way to send messages fully synchronously so we're guaranteed that when the function returns all the messages have been processed?

Reply:
The sleep is unavoidable for the following reasons
- these messages are sent between  2 running libp2p nodes we don't have any hooks in between
- these are messages sent _after_ the node is blocked meaning that these messages are not expected to be delivered to the receiver node, so we sleep this approximate amount of time to ensure all messages were attempted, processed and dropped

Reply:
Can you add this sleep justification to the test?

Reply:
https://github.com/onflow/flow-go/pull/3877/commits/51696434cfb484239a4a2351a603f83a1be71d9e